core-bundle/src/Image/Studio/FigureBuilder.php

```diff
@@ -0,0 +1,595 @@
+                }, $settings
+            ),
+            \Closure::bind(
+                function (Figure $figure): ?LightBoxResult {
+                    return $this->onDefineLightBoxResult($figure);
+                }, $settings
+            )
+        );
+    }
+
+    /**
+     * Define meta data [on demand].
+     */
+    private function onDefineMetaData(): ?MetaData
+    {
+        if ($this->disableMetaData) {
+            return null;
+        }
+
+        if (null !== $this->metaData) {
+            return $this->metaData;
+        }
+
+        if (null === $this->filesModel) {
+            return null;
+        }
+
+        // Get fallback locale list or use without fallbacks if explicitly set
+        $locales = null !== $this->locale ? [$this->locale] : $this->getFallbackLocaleList();
+
+        $metaData = $this->filesModel->getMetaData(...$locales);
+
+        if (null !== $metaData) {
+            return $metaData;
+        }
+
+        // If no meta data can be obtained from the model, we create a
+        // container from the default meta fields with empty values instead.
+        $metaFields = $this->filesModelAdapter()->getMetaFields();
+
+        return new MetaData(
+            array_combine($metaFields, array_fill(0, \count($metaFields), ''))
+        );
+    }
+
+    /**
+     * Define link attributes [on demand].
+     */
+    private function onDefineLinkAttributes(Figure $result): array
+    {
+        $linkAttributes = [];
+
+        // Open in a new window if light box was requested but is invalid (fullsize)
+        if ($this->enableLightBox && !$result->hasLightBox()) {
+            $linkAttributes['target'] = '_blank';
+        }
+
+        return array_merge($linkAttributes, $this->additionalLinkAttributes);
+    }
+
+    /**
+     * Define the light box result if it is enabled [on demand].
+     */
+    private function onDefineLightBoxResult(Figure $result): ?LightBoxResult
+    {
+        if (!$this->enableLightBox) {
+            return null;
+        }
+
+        $getMetaDataUrl = static function () use ($result): ?string {
+            if (!$result->hasMetaData()) {
+                return null;
+            }
+
+            return $result->getMetaData()->getUrl() ?: null;
+        };
+
+        $getResourceOrUrl = function ($target): array {
+            if ($target instanceof ImageInterface) {
+                return [$target, null];
+            }
+
+            $validExtensions = $this->locator
+                ->get('parameter_bag')
+                ->get('contao.image.valid_extensions')
+            ;
+
+            $validExtension = \in_array(Path::getExtension($target), $validExtensions, true);
+            $externalUrl = 1 === preg_match('#^https?://#', $target);
+
+            if ($externalUrl) {
+                return $validExtension ? [null, $target] : [null, null];
+            }
+
+            if (!$validExtension) {
+                return [null, $target];
+            }
```

Comment:
This seems to be wrong. Shouldn’t it be something like this?
```suggestion
            if (!$validExtension) {
                return [null, null];
            }

            if ($externalUrl) {
                return [null, $target];
            }
```
In my test setup internal links to HTML-pages now show up as lightbox links instead of `target=_blank` links... This change would fix this.

Reply:
I guess you're right. This logic confuses me. :zany_face: 

This would mean the following, right?

| URL      | Extension | Lightbox            | Link Attributes                           |
|----------|-----------|---------------------|-------------------------------------------|
| internal | valid     | yes (resized image) |                                           |
| internal | invalid   | no                  | target="_blank"                           |
| external | valid     | yes (URL)           | rel="noreferrer noopener"                 |
| external | invalid   | no                  | target="_blank" rel="noreferrer noopener" |

Reply:
Adjusted + added tests in 0c777cef4b281c7a67618ad7750802339de8f497.